### PR TITLE
Avoid str cast causing ascii bug.

### LINKED ISSUE #2803: Unable to Delete Old Inactive User
* Specific code commit: f36dfda3e6f60a24b2372d32fa34a0a8d9365470
* Version of the web browser and OS: Version 66.0.3359.181 
* Environment where the error occurred (Vagrant VM and site conf mode or AWS instance): 
beta site

### Steps to reproduce
Please list all the actions and the input data used:

1. Admin interface
2. Try to delete user #42

### Observed behavior
```
Internal Server Error: /admin/auth/user/42/delete/
Traceback (most recent call last):
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/contrib/admin/options.py", line 618, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/utils/decorators.py", line 110, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/views/decorators/cache.py", line 57, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/contrib/admin/sites.py", line 233, in inner
    return view(request, *args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/utils/decorators.py", line 34, in _wrapper
    return bound_func(*args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/utils/decorators.py", line 110, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/utils/decorators.py", line 30, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/contrib/admin/options.py", line 1709, in delete_view
    [obj], opts, request.user, self.admin_site, using)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/contrib/admin/utils.py", line 153, in get_deleted_objects
    to_delete = collector.nested(format_callback)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/contrib/admin/utils.py", line 213, in nested
    roots.extend(self._nested(root, seen, format_callback))
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/contrib/admin/utils.py", line 196, in _nested
    children.extend(self._nested(child, seen, format_callback))
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/contrib/admin/utils.py", line 198, in _nested
    ret = [format_callback(obj)]
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/contrib/admin/utils.py", line 126, in format_callback
    force_text(obj))
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/utils/encoding.py", line 92, in force_text
    s = six.text_type(s)
  File "/srv/refinery-platform/refinery/core/models.py", line 98, in __unicode__
    str(self.affiliation) + "): " +
UnicodeEncodeError: 'ascii' codec can't encode character u'\u0443' in position 15: ordinal not in range(128)

Request repr(): 
<WSGIRequest
path:/admin/auth/user/42/delete/,
GET:<QueryDict: {}>,
POST:<QueryDict: {}>,
META:{'CONTEXT_DOCUMENT_ROOT': '/var/www',
 'CONTEXT_PREFIX': '',
 'DOCUMENT_ROOT': '/var/www',
 'GATEWAY_INTERFACE': 'CGI/1.1',
 'HTTP_ACCEPT': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8',
 'HTTP_ACCEPT_ENCODING': 'gzip, deflate, br',
 'HTTP_ACCEPT_LANGUAGE': 'en-US,en;q=0.9',
 'HTTP_CONNECTION': 'keep-alive',
 'HTTP_HOST': 'beta.stemcellcommons.org',
 'HTTP_REFERER': 'https://beta.stemcellcommons.org/admin/auth/user/42/',
 'HTTP_UPGRADE_INSECURE_REQUESTS': '1',
 'HTTP_USER_AGENT': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.181 Safari/537.36',
 'HTTP_X_FORWARDED_FOR': '134.174.140.196',
 'HTTP_X_FORWARDED_PORT': '443',
 'HTTP_X_FORWARDED_PROTO': 'https',
 'PATH_INFO': u'/admin/auth/user/42/delete/',
 'PATH_TRANSLATED': '/srv/refinery-platform/refinery/config/wsgi_aws.py/admin/auth/user/42/delete/',
 'QUERY_STRING': '',
 'REMOTE_ADDR': '172.30.0.39',
 'REMOTE_PORT': '42824',
 'REQUEST_METHOD': 'GET',
 'REQUEST_SCHEME': 'http',
 'REQUEST_URI': '/admin/auth/user/42/delete/',
 'SCRIPT_FILENAME': '/srv/refinery-platform/refinery/config/wsgi_aws.py',
 'SCRIPT_NAME': u'',
 'SCRIPT_URI': 'http://beta.stemcellcommons.org/admin/auth/user/42/delete/',
 'SCRIPT_URL': '/admin/auth/user/42/delete/',
 'SERVER_ADDR': '172.30.0.130',
 'SERVER_ADMIN': '[no address given]',
 'SERVER_NAME': 'beta.stemcellcommons.org',
 'SERVER_PORT': '80',
 'SERVER_PROTOCOL': 'HTTP/1.1',
 'SERVER_SIGNATURE': '',
 'SERVER_SOFTWARE': 'Apache/2.4.7 (Ubuntu)',
 'mod_wsgi.application_group': 'beta.stemcellcommons.org|',
 'mod_wsgi.callable_object': 'application',
 'mod_wsgi.enable_sendfile': '0',
 'mod_wsgi.handler_script': '',
 'mod_wsgi.input_chunked': '0',
 'mod_wsgi.listener_host': '',
 'mod_wsgi.listener_port': '80',
 'mod_wsgi.process_group': 'refinery',
 'mod_wsgi.queue_start': '1528131430976405',
 'mod_wsgi.request_handler': 'wsgi-script',
 'mod_wsgi.script_reloading': '1',
 'mod_wsgi.version': (3, 4),
 'wsgi.errors': <mod_wsgi.Log object at 0x7f02c90db7f0>,
 'wsgi.file_wrapper': <built-in method file_wrapper of mod_wsgi.Adapter object at 0x7f02c95fd468>,
 'wsgi.input': <mod_wsgi.Input object at 0x7f02c9180c70>,
 'wsgi.multiprocess': False,
 'wsgi.multithread': True,
 'wsgi.run_once': False,
 'wsgi.url_scheme': 'http',
 'wsgi.version': (1, 0)}>
```

### Expected behavior
* To delete it

### Notes
* Related to #2802 
* Probable root cause: #2557

### PR DESCRIPTION
**Pull request by jkmarx**
- Resolves #2803 

-~ [ ] Will write a separate technical debt issue, should avoid str casting in unicode~
-~ [ ] Upgrade unicode to str (see comments below)~

Ref: #2619

screen shot 2018-07-13 at 2 13 13 pm

delete_user

### CODE REVIEW

File: refinery/core/models.py [hunk 1]
```diff
@@ -97,10 +97,10 @@ class UserProfile(models.Model):
...
     def __unicode__(self):
         return (
-            str(self.user.first_name) + " " +
-            str(self.user.last_name) + " (" +
-            str(self.affiliation) + "): " +
-            str(self.user.email)
+            self.user.first_name + " " +
+            self.user.last_name + " (" +
+            self.affiliation + "): " +
+            self.user.email
```
Comment (hackdna) [inline refinery/core/models.py]: why not make it a bit more compact?
```python
        return (self.user.first_name + " " + self.user.last_name +
                " (" + self.affiliation + "): " + self.user.email)
```

**Comment by hackdna** [comment]
I'd also replace `__unicode__` with `__str__` to help with transition to Python 3: https://docs.djangoproject.com/en/1.8/topics/python3/#str-and-unicode-methods

File: refinery/core/models.py [hunk 2]
```diff
@@ -95,13 +95,11 @@ class UserProfile(models.Model):
...
     login_count = models.IntegerField(default=0)
 
-    def __unicode__(self):
-        return (
-            str(self.user.first_name) + " " +
-            str(self.user.last_name) + " (" +
-            str(self.affiliation) + "): " +
-            str(self.user.email)
-        )
+    def __str__(self):
```
Comment (hackdna) [inline refinery/core/models.py]: `python_2_unicode_compatible()` decorator should be added:
https://docs.djangoproject.com/en/1.8/topics/python3/#str-and-unicode-methods

Comment (jkmarx) [reply refinery/core/models.py]: Yep, "Django provides a simple way to define __str__() and ` __unicode__()`_ methods that work on Python 2 and 3: you must define a __str__() method returning text and to apply the python_2_unicode_compatible() decorator."

Comment (hackdna) [reply refinery/core/models.py]: Right, so where's the decorator? :)

Comment (jkmarx) [reply refinery/core/models.py]: @hackdna For future reference, I'll tag or msg you when a pull request is ready for another round of review.

**Comment by jkmarx** [comment]
http://python-future.org/unicode_literals.html

"In general, it is more compelling to use unicode_literals when back-porting new or existing Python 3 code to Python 2/3 than when porting existing Python 2 code to 2/3. In the latter case, explicitly marking up all unicode string literals with u'' prefixes would help to avoid unintentionally changing the existing Python 2 API. However, if changing the existing Python 2 API is not a concern, using unicode_literals may speed up the porting process."

**Comment by jkmarx** [comment]
@hackdna Replace __unicode__ with __str__ will not be included in this pull request, I'll include it in the upgrade issue.

**Pull request merged**